### ISSUE TITLE
Some numeric conversions generate uncompilable rust code 

### ISSUE THREAD

**Issue by mbudiu-vmw** [vmware/differential-datalog#563]
```
use fp
function ftoi(l: double): signed<32> = {
    match (int_from_d(l)) {
        None -> 0,
        Some{x} -> x as signed<32>
    }
}

relation Rdummy(x: Tnumber)
Rdummy(0).
relation RFloatToInt(_x:Tnumber)
RFloatToInt(_x) :- Rdummy(0), var _x = ftoi((- (64'f333.36: double))).
```

This produces:

```
error[E0369]: binary operation `&` cannot be applied to type `differential_datalog::int::Int`
    --> types/lib.rs:2368:54
     |
2368 |         std_Option::std_Some{x: ref x} => (x.clone() & ((Int::one() <<32) - Int::one())).to_i32().unwrap()
     |                                            --------- ^ -------------------------------- differential_datalog::int::Int
     |                                            |
     |                                            differential_datalog::int::Int
     |
     = note: an implementation of `std::ops::BitAnd` might be missing for `differential_datalog::int::Int`
```

**Comment by mbudiu-vmw** [comment]
The relation may not even be needed.

**Comment by ryzhyk** [comment]
I was able to reproduce

**Comment by mbudiu-vmw** [comment]
That's probably the code I checked in for the double support, but I didn't have time to fix it now.

**Comment by ryzhyk** [comment]
Closed by #565
